### ISSUE TITLE
`AndroidOverscrollIndicator.stretch` freezes and throws error in `NestedScrollView`

### ISSUE THREAD

**Issue by Mayb3Nots** [flutter/flutter#100967]
## Steps to Reproduce

1. Run example code below
2. Drag the listview to the bottom letting it overscroll.
3. After a few attempt it should freezes and throw a " build scheduled during frame"  error

**Expected results:** 
Should not freeze or throw an error.

**Actual results:** 
ListView freezes and throws an error.

<details>

<summary>Flutter doctor</summary>

```
[√] Flutter (Channel stable, 2.10.3, on Microsoft Windows [Version 10.0.22000.556], locale en-MY)
[√] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
[√] Chrome - develop for the web
[X] Visual Studio - develop for Windows
    X Visual Studio not installed; this is necessary for Windows development.
      Download at https://visualstudio.microsoft.com/downloads/.
      Please install the "Desktop development with C++" workload, including all of its default components
[√] Android Studio (version 2021.1)
[√] VS Code (version 1.65.2)
[√] Connected device (4 available)
[√] HTTP Host Availability
```
</details>
<details>
<summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({Key? key}) : super(key: key);
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
         useMaterial3: true,
       ),
      home: Scaffold(
        body: NestedScrollView(
          headerSliverBuilder: (context, innerBoxIsScrolled) => [
            const SliverAppBar(
              title: Text('Overscroll bug with NestedScrollView'),
            )
          ],
          body: ListView(
              padding: EdgeInsets.zero,
              children: List.generate(
                  100,
                  (index) => Container(
                      height: 100,
                      color: index.isEven ? Colors.amber : Colors.orange,
                      child: Text(index.toString())))),
        ),
      ),
    );
  }
}
```

</details>

<details>
  <summary>Logs</summary>

```
════════ Exception caught by foundation library ════════════════════════════════
The following assertion was thrown while dispatching notifications for _StretchController:
Build scheduled during frame.

While the widget tree was being built, laid out, and painted, a new frame was scheduled to rebuild the widget tree.

This might be because setState() was called from a layout or paint callback. If a change is needed to the widget tree, it should be applied as the tree is being built. Scheduling a change for the subsequent frame instead results in an interface that lags behind by one frame. If this was done to make your build dependent on a size measured at layout time, consider using a LayoutBuilder, CustomSingleChildLayout, or CustomMultiChildLayout. If, on the other hand, the one frame delay is the desired effect, for example because this is an animation, consider scheduling the frame in a post-frame callback using SchedulerBinding.addPostFrameCallback or using an AnimationController to trigger the animation.

When the exception was thrown, this was the stack
#0      WidgetsBinding._handleBuildScheduled.<anonymous closure>
package:flutter/…/widgets/binding.dart:747
#1      WidgetsBinding._handleBuildScheduled
package:flutter/…/widgets/binding.dart:770
#2      BuildOwner.scheduleBuildFor
package:flutter/…/widgets/framework.dart:2485
#3      Element.markNeedsBuild
package:flutter/…/widgets/framework.dart:4443
#4      State.setState
package:flutter/…/widgets/framework.dart:1141
#5      _AnimatedState._handleChange
package:flutter/…/widgets/transitions.dart:128
#6      ChangeNotifier.notifyListeners
package:flutter/…/foundation/change_notifier.dart:308
#7      AnimationLocalListenersMixin.notifyListeners
package:flutter/…/animation/listener_helpers.dart:155
#8      AnimationController.value=
package:flutter/…/animation/animation_controller.dart:365
#9      AnimationController.forward
package:flutter/…/animation/animation_controller.dart:460
#10     _StretchController._recede
package:flutter/…/widgets/overscroll_indicator.dart:895
#11     _StretchController.scrollEnd
package:flutter/…/widgets/overscroll_indicator.dart:869
#12     _StretchingOverscrollIndicatorState._handleScrollNotification
package:flutter/…/widgets/overscroll_indicator.dart:715
#13     NotificationListener._dispatch
package:flutter/…/widgets/notification_listener.dart:151
#14     Notification.visitAncestor
package:flutter/…/widgets/notification_listener.dart:67
#15     ViewportNotificationMixin.visitAncestor
package:flutter/…/widgets/scroll_notification.dart:31
#16     Element.visitAncestorElements
package:flutter/…/widgets/framework.dart:4254
#17     Notification.dispatch
package:flutter/…/widgets/notification_listener.dart:83
#18     ScrollActivity.dispatchScrollEndNotification
package:flutter/…/widgets/scroll_activity.dart:104
#19     ScrollPosition.didEndScroll
package:flutter/…/widgets/scroll_position.dart:907
#20     ScrollPosition.beginActivity
package:flutter/…/widgets/scroll_position.dart:876
#21     _NestedInnerBallisticScrollActivity.applyNewDimensions
package:flutter/…/widgets/nested_scroll_view.dart:1449
#22     ScrollPosition.applyNewDimensions
package:flutter/…/widgets/scroll_position.dart:623
#23     _NestedScrollPosition.applyNewDimensions
package:flutter/…/widgets/nested_scroll_view.dart:1405
#24     ScrollPosition.applyContentDimensions
package:flutter/…/widgets/scroll_position.dart:553
#25     RenderViewport.performLayout
package:flutter/…/rendering/viewport.dart:1493
#26     RenderObject.layout
package:flutter/…/rendering/object.dart:1887
#27     RenderProxyBoxMixin.performLayout
package:flutter/…/rendering/proxy_box.dart:116
#28     RenderObject.layout
package:flutter/…/rendering/object.dart:1887
#29     RenderProxyBoxMixin.performLayout
package:flutter/…/rendering/proxy_box.dart:116
#30     RenderObject.layout
package:flutter/…/rendering/object.dart:1887
#31     RenderProxyBoxMixin.performLayout
package:flutter/…/rendering/proxy_box.dart:116
#32     RenderObject.layout
package:flutter/…/rendering/object.dart:1887
#33     RenderProxyBoxMixin.performLayout
package:flutter/…/rendering/proxy_box.dart:116
#34     RenderObject.layout
package:flutter/…/rendering/object.dart:1887
#35     RenderProxyBoxMixin.performLayout
package:flutter/…/rendering/proxy_box.dart:116
#36     RenderObject.layout
package:flutter/…/rendering/object.dart:1887
#37     RenderProxyBoxMixin.performLayout
package:flutter/…/rendering/proxy_box.dart:116
#38     RenderObject.layout
package:flutter/…/rendering/object.dart:1887
#39     RenderProxyBoxMixin.performLayout
package:flutter/…/rendering/proxy_box.dart:116
#40     RenderObject.layout
package:flutter/…/rendering/object.dart:1887
#41     RenderProxyBoxMixin.performLayout
package:flutter/…/rendering/proxy_box.dart:116
#42     _RenderCustomClip.performLayout
package:flutter/…/rendering/proxy_box.dart:1376
#43     RenderObject.layout
package:flutter/…/rendering/object.dart:1887
#44     RenderSliverFillRemainingWithScrollable.performLayout
package:flutter/…/rendering/sliver_fill.dart:92
#45     RenderObject.layout
package:flutter/…/rendering/object.dart:1887
#46     RenderViewportBase.layoutChildSequence
package:flutter/…/rendering/viewport.dart:510
#47     RenderViewport._attemptLayout
package:flutter/…/rendering/viewport.dart:1580
#48     RenderViewport.performLayout
package:flutter/…/rendering/viewport.dart:1489
#49     RenderObject._layoutWithoutResize
package:flutter/…/rendering/object.dart:1731
#50     PipelineOwner.flushLayout
package:flutter/…/rendering/object.dart:887
#51     RendererBinding.drawFrame
#52     WidgetsBinding.drawFrame
package:flutter/…/widgets/binding.dart:883
#53     RendererBinding._handlePersistentFrameCallback
package:flutter/…/rendering/binding.dart:363
#54     SchedulerBinding._invokeFrameCallback
package:flutter/…/scheduler/binding.dart:1144
#55     SchedulerBinding.handleDrawFrame
package:flutter/…/scheduler/binding.dart:1081
#56     SchedulerBinding._handleDrawFrame
package:flutter/…/scheduler/binding.dart:995
#60     _invoke (dart:ui/hooks.dart:151:10)
#61     PlatformDispatcher._drawFrame (dart:ui/platform_dispatcher.dart:308:5)
#62     _drawFrame (dart:ui/hooks.dart:115:31)
(elided 3 frames from dart:async)
The _StretchController sending notification was: _StretchController()
════════════════════════════════════════════════════════════════════════════════


```

```
```

</details>

https://user-images.githubusercontent.com/68549184/160637983-ffbce7a8-a993-4e5d-ac9d-d15b19af9845.mp4

In a side note in my app it happens only if you spam drag it and it hits a certain "stretchyness" . However it worth noting that my implementation of `NestedScrollView` is different from the sample code. I also used `CustomScrollView` with `SliverOverlapInjector/SliverOverlapAbsorber` and a `SliverPersistenHeader` for the bottom part of the video.

https://user-images.githubusercontent.com/68549184/160638691-f3dacb7a-b31d-4d0b-9fa4-c83b6687f164.mp4

**Comment by darshankawar** [comment]
@Mayb3Nots 
Although I do get same behavior as you running on Android, but I verified this on master and there it seems the property `androidOverscrollIndicator` is deprecated:

<img width="816" alt="Screenshot 2022-03-30 at 12 03 07 PM" src="https://user-images.githubusercontent.com/67046386/160766591-923c23c3-5aa4-4348-a4b2-456f0b72ac3b.png">

Can you try to use the suggestion as mentioned in the info and see if it helps ?

**Comment by Mayb3Nots** [comment]
Hi @darshankawar I can confirm that on my sample code after using the `userMaterial3` and removing the `andriodOverscrollIndicator `with master branch will still reproduce the same issue.

**Comment by darshankawar** [comment]
Thanks for the test and update. I do see the same behavior on latest master and stable.

<details>
<summary> console error log </summary>

```

════════ Exception caught by foundation library ════════════════════════════════
The following assertion was thrown while dispatching notifications for _StretchController:
Build scheduled during frame.

While the widget tree was being built, laid out, and painted, a new frame was scheduled to rebuild the widget tree.

This might be because setState() was called from a layout or paint callback. If a change is needed to the widget tree, it should be applied as the tree is being built. Scheduling a change for the subsequent frame instead results in an interface that lags behind by one frame. If this was done to make your build dependent on a size measured at layout time, consider using a LayoutBuilder, CustomSingleChildLayout, or CustomMultiChildLayout. If, on the other hand, the one frame delay is the desired effect, for example because this is an animation, consider scheduling the frame in a post-frame callback using SchedulerBinding.addPostFrameCallback or using an AnimationController to trigger the animation.

When the exception was thrown, this was the stack
#0      WidgetsBinding._handleBuildScheduled.<anonymous closure>
package:flutter/…/widgets/binding.dart:747
#1      WidgetsBinding._handleBuildScheduled
package:flutter/…/widgets/binding.dart:770
#2      BuildOwner.scheduleBuildFor
package:flutter/…/widgets/framework.dart:2485
#3      Element.markNeedsBuild
package:flutter/…/widgets/framework.dart:4443
#4      State.setState
package:flutter/…/widgets/framework.dart:1141
#5      _AnimatedState._handleChange
package:flutter/…/widgets/transitions.dart:128
#6      ChangeNotifier.notifyListeners
package:flutter/…/foundation/change_notifier.dart:308
#7      AnimationLocalListenersMixin.notifyListeners
package:flutter/…/animation/listener_helpers.dart:155
#8      AnimationController.value=
package:flutter/…/animation/animation_controller.dart:365
#9      AnimationController.forward
package:flutter/…/animation/animation_controller.dart:460
#10     _StretchController._recede
package:flutter/…/widgets/overscroll_indicator.dart:895
#11     _StretchController.scrollEnd
package:flutter/…/widgets/overscroll_indicator.dart:869
#12     _StretchingOverscrollIndicatorState._handleScrollNotification
package:flutter/…/widgets/overscroll_indicator.dart:715
#13     NotificationListener._dispatch
package:flutter/…/widgets/notification_listener.dart:151
#14     Notification.visitAncestor
package:flutter/…/widgets/notification_listener.dart:67
#15     ViewportNotificationMixin.visitAncestor
package:flutter/…/widgets/scroll_notification.dart:31
#16     Element.visitAncestorElements
package:flutter/…/widgets/framework.dart:4254
#17     Notification.dispatch
package:flutter/…/widgets/notification_listener.dart:83
#18     ScrollActivity.dispatchScrollEndNotification
package:flutter/…/widgets/scroll_activity.dart:104
#19     ScrollPosition.didEndScroll
package:flutter/…/widgets/scroll_position.dart:907
#20     ScrollPosition.beginActivity
package:flutter/…/widgets/scroll_position.dart:876
#21     _NestedInnerBallisticScrollActivity.applyNewDimensions
package:flutter/…/widgets/nested_scroll_view.dart:1449
#22     ScrollPosition.applyNewDimensions
package:flutter/…/widgets/scroll_position.dart:623
#23     _NestedScrollPosition.applyNewDimensions
package:flutter/…/widgets/nested_scroll_view.dart:1405
#24     ScrollPosition.applyContentDimensions
package:flutter/…/widgets/scroll_position.dart:553
#25     RenderViewport.performLayout
package:flutter/…/rendering/viewport.dart:1493
#26     RenderObject.layout
package:flutter/…/rendering/object.dart:1887
#27     RenderProxyBoxMixin.performLayout
package:flutter/…/rendering/proxy_box.dart:116
#28     RenderObject.layout
package:flutter/…/rendering/object.dart:1887
#29     RenderProxyBoxMixin.performLayout
package:flutter/…/rendering/proxy_box.dart:116
#30     RenderObject.layout
package:flutter/…/rendering/object.dart:1887
#31     RenderProxyBoxMixin.performLayout
package:flutter/…/rendering/proxy_box.dart:116
#32     RenderObject.layout
package:flutter/…/rendering/object.dart:1887
#33     RenderProxyBoxMixin.performLayout
package:flutter/…/rendering/proxy_box.dart:116
#34     RenderObject.layout
package:flutter/…/rendering/object.dart:1887
#35     RenderProxyBoxMixin.performLayout
package:flutter/…/rendering/proxy_box.dart:116
#36     RenderObject.layout
package:flutter/…/rendering/object.dart:1887
#37     RenderProxyBoxMixin.performLayout
package:flutter/…/rendering/proxy_box.dart:116
#38     RenderObject.layout
package:flutter/…/rendering/object.dart:1887
#39     RenderProxyBoxMixin.performLayout
package:flutter/…/rendering/proxy_box.dart:116
#40     RenderObject.layout
package:flutter/…/rendering/object.dart:1887
#41     RenderProxyBoxMixin.performLayout
package:flutter/…/rendering/proxy_box.dart:116
#42     _RenderCustomClip.performLayout
package:flutter/…/rendering/proxy_box.dart:1376
#43     RenderObject.layout
package:flutter/…/rendering/object.dart:1887
#44     RenderSliverFillRemainingWithScrollable.performLayout
package:flutter/…/rendering/sliver_fill.dart:92
#45     RenderObject.layout
package:flutter/…/rendering/object.dart:1887
#46     RenderViewportBase.layoutChildSequence
package:flutter/…/rendering/viewport.dart:510
#47     RenderViewport._attemptLayout
package:flutter/…/rendering/viewport.dart:1580
#48     RenderViewport.performLayout
package:flutter/…/rendering/viewport.dart:1489
#49     RenderObject._layoutWithoutResize
package:flutter/…/rendering/object.dart:1731
#50     PipelineOwner.flushLayout
package:flutter/…/rendering/object.dart:887
#51     RendererBinding.drawFrame
#52     WidgetsBinding.drawFrame
package:flutter/…/widgets/binding.dart:883
#53     RendererBinding._handlePersistentFrameCallback
package:flutter/…/rendering/binding.dart:363
#54     SchedulerBinding._invokeFrameCallback
package:flutter/…/scheduler/binding.dart:1144
#55     SchedulerBinding.handleDrawFrame
package:flutter/…/scheduler/binding.dart:1081
#56     SchedulerBinding._handleDrawFrame
package:flutter/…/scheduler/binding.dart:995
#60     _invoke (dart:ui/hooks.dart:151:10)
#61     PlatformDispatcher._drawFrame (dart:ui/platform_dispatcher.dart:308:5)
#62     _drawFrame (dart:ui/hooks.dart:115:31)
(elided 3 frames from dart:async)
The _StretchController sending notification was: _StretchController()
════════════════════════════════════════════════════════════════════════════════


```
</details>

<details>
<summary> stable, master flutter doctor -v </summary>

```

[✓] Flutter (Channel stable, 2.10.4, on macOS 12.2.1 21D62 darwin-x64, locale
    en-GB)
    • Flutter version 2.10.4 at /Users/dhs/documents/fluttersdk/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision c860cba910 (3 days ago), 2022-03-25 00:23:12 -0500
    • Engine revision 57d3bac3dd
    • Dart version 2.16.2
    • DevTools version 2.9.2

[!] Xcode - develop for iOS and macOS (Xcode 12.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    ! Flutter recommends a minimum Xcode version of 13.
      Download the latest version or update via the Mac App Store.
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] VS Code (version 1.62.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.21.0

[✓] Connected device (5 available)
    • SM G975F (mobile)       • RZ8M802WY0X • android-arm64   • Android 11 (API 30)
    • Darshan's iphone (mobile)  • 21150b119064aecc249dfcfe05e259197461ce23 •
      ios            • iOS 14.4.1 18D61
    • iPhone 12 Pro Max (mobile) • A5473606-0213-4FD8-BA16-553433949729     •
      ios            • com.apple.CoreSimulator.SimRuntime.iOS-14-3 (simulator)
    • macOS (desktop)            • macos                                    •
      darwin-x64     • Mac OS X 10.15.4 19E2269 darwin-x64
    • Chrome (web)               • chrome                                   •
      web-javascript • Google Chrome 98.0.4758.80

[✓] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 1 category.

[[✓] Flutter (Channel master, 2.13.0-0.0.pre.310, on macOS 12.2.1 21D62
    darwin-x64, locale en-GB)
    • Flutter version 2.13.0-0.0.pre.310 at
      /Users/dhs/documents/fluttersdk/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 255f967ca2 (60 minutes ago), 2022-03-29 20:30:13 -0700
    • Engine revision 0e7773bcca
    • Dart version 2.17.0 (build 2.17.0-249.0.dev)
    • DevTools version 2.11.4

[!] Xcode - develop for iOS and macOS (Xcode 12.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    ! Flutter recommends a minimum Xcode version of 13.
      Download the latest version or update via the Mac App Store.
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] VS Code (version 1.62.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.21.0

[✓] Connected device (5 available)
    • SM G975F (mobile)       • RZ8M802WY0X • android-arm64   • Android 11 (API 30)
    • Darshan's iphone (mobile)  • 21150b119064aecc249dfcfe05e259197461ce23 •
      ios            • iOS 14.4.1 18D61
    • iPhone 12 Pro Max (mobile) • A5473606-0213-4FD8-BA16-553433949729     •
      ios            • com.apple.CoreSimulator.SimRuntime.iOS-14-3 (simulator)
    • macOS (desktop)            • macos                                    •
      darwin-x64     • Mac OS X 10.15.4 19E2269 darwin-x64
    • Chrome (web)               • chrome                                   •
      web-javascript • Google Chrome 98.0.4758.80

[✓] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 1 category.



```

</details>

**Comment by Mayb3Nots** [comment]
closed in favour of #104798
